### PR TITLE
HSEARCH-4034 Add a multi-tenancy test to mapper ORM integration tests

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-4034

### CODE REVIEW

File: integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/multitenancy/DatabaseMultitenancyIT.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+	@Test
+	public void searchReferences() {
+		List<EntityReference> entityReferences = searchReferences( GEOCHRON_TID, GEOCHRON_MODELS );
+		assertThat( entityReferences ).hasSize( GEOCHRON_MODELS.length );
+	}
+
+	private void persist(String tenantId, Clock[] models) {
+		BackendMock.DocumentWorkCallListContext expectWorks = backendMock.expectWorks( Clock.INDEX );
+		for ( Clock clock : models ) {
+			expectWorks.add( clock.getId() + "", tenantId, b -> b.field( "brand", clock.getBrand() ) );
```
Comment (yrodiere) [inline integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/multitenancy/DatabaseMultitenancyIT.java:106]: Wouldn't it make more sense to pass the tenantId to `expectWorks()`, just above the `for` loop? That's what we do for `expectIndexScaleWorks`, at least.

**Review by yrodiere — CHANGES_REQUESTED** [review]
Just one small comment, apart from that +1 to merge.

And yes, I'll look into the test failure, which is unrelated to your changes.

File: integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/multitenancy/DatabaseMultitenancyIT.java (hunk 1 above)
Comment (fax4ever) [reply integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/multitenancy/DatabaseMultitenancyIT.java]: Yes it is better to set the tenant on `DocumentWorkCallListContext` once and for all

**Comment by fax4ever** [comment]
@yrodiere thank you for the review, I updated the pull request with a new commit for the fix.

**Comment by yrodiere** [comment]
I amended your commit to fix the ticket key (it was wrong) and make sure that the tenantId is used for all operations (including delete). Feel free to merge once the build passes, thanks!

**Comment by fax4ever** [comment]
I pulled the commit amend, then re-based on top of master, I'm going to push force.

**Comment by fax4ever** [comment]
I'll wait for the CI then I merge it.

**Comment by fax4ever** [comment]
Merged thanks!

**Pull request merged**